### PR TITLE
Office user alerted of infected file

### PR DESCRIPTION
**Pull request by ralren**
## Description

- Can now fetch tags from S3
- New property called `tags` on the uploads payload (not the model)
- Alert pops up if upload tag indicates file is infected

## Setup

```sh
make clean
rm -rf pkg/gen
make server_generate
env STORAGE_BACKEND=s3 make server_run_standalone
make client_run
make office_client_run
```

1. Make a new move.
2. Find that move in the office side.
3. Upload a new document.
4. Navigate to the document and confirm it shows up as normal. This might take a minute or two.
5. Login to AWS S3
6. Navigate to S3 > transcom-ppp-app-devlocal-us-west-2 > (folder with your name) > user > (folder with office user uuid) > uploads > (only file that should be there) > properties
7. Within the tags, add a tag `av-notes` that you personally changed this on this day. Change `av-status` from `CLEAN` to `INFECTED`. NOTE: You will not be able to modify this upload anymore.
8. Navigate to the document in office. Confirm that an alert shows up instead of the upload.

## Code Review Verification Steps
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/168412033) for this change

## Screenshots
Screen Shot 2019-10-16 at 3 26 20 PM

### CODE REVIEW

File: src/shared/DocumentViewer/DocumentContent.jsx [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
 import styles from './DocumentContent.module.scss';
 import { RotationBar } from 'shared/RotationBar/RotationBar';
+import Alert from 'shared/Alert';
 
 const DocumentContent = props => {
-  let { contentType, filename, url } = props;
+  let { contentType, filename, url, tags } = props;
+  let isInfected = false;
+  forEach(tags, function(tag) {
+    if (tag.key === 'av-status' && tag.value === 'INFECTED') {
```
Comment (chrisgilmerproj) [inline src/shared/DocumentViewer/DocumentContent.jsx]: Can we make these constants? `av-status` and `INFECTED`? I feel like they'll get re-used somewhere and I'd like to not have a mispelling error. The other thing is we may eventually want to change this to an env var.

File: src/shared/DocumentViewer/DocumentContent.test.jsx [hunk 2]
```diff
@@ -13,6 +14,17 @@ describe('DocumentContent', () => {
       const wrapper = shallow(<DocumentContent contentType="image/jpeg" url="www" filename="filename" />);
       expect(wrapper.find(NonPDFImage)).toHaveLength(1);
     });
+    it('renders an Alert when tags indicate document is infected', () => {
+      const wrapper = shallow(
+        <DocumentContent
+          contentType="application/pdf"
+          url="www"
+          filename="filename"
+          tags={[{ key: 'av-status', value: 'INFECTED' }]}
```
Comment (chrisgilmerproj) [inline src/shared/DocumentViewer/DocumentContent.test.jsx]: Use the constants here.

**Comment by ralren** [comment]
@chrisgilmerproj you've made excellent points. We're planning to head in that direction with another ticket if that's satisfactory.

**Comment by chrisgilmerproj** [comment]
> @chrisgilmerproj you've made excellent points. We're planning to head in that direction with another ticket if that's satisfactory.

My concern is mostly about leaking info. But if we're going to fix that in another PR or this PR then I'm fine.

**Comment by mkrump** [comment]
> @chrisgilmerproj you've made excellent points. We're planning to head in that direction with another ticket if that's satisfactory.

Do you mind linking to the ticket? I think @chrisgilmerproj suggestion not to expose any tags we don't have to is a good one. Just wanted to make sure that it's recored somewhere in case you aren't the one to pickup the work.

**Comment by ralren** [comment]
@mkrump here you go https://www.pivotaltracker.com/story/show/169216570

File: pkg/storage/s3.go [hunk 3]
```diff
@@ -123,3 +123,26 @@ func (s *S3) PresignedURL(key string, contentType string) (string, error) {
...
+		Key:    &namespacedKey,
+	}
+
+	result, err := s.client.GetObjectTagging(input)
+	if err != nil {
+		return tags, errors.Wrap(err, "get object tagging on s3 failed")
+	}
+
+	for _, tag := range result.TagSet {
+		tags[*tag.Key] = *tag.Value
```
Comment (mkrump) [inline pkg/storage/s3.go]: do you have to do a nil check here? I think looks like not looking at the s3 sdk defn.

**Pull request merged**